controllers/mytba_controller.py

```diff
@@ -19,21 +21,38 @@ def get(self):
         self._require_registration('/account/register')
 
         user = self.user_bundle.account.key
+        now = datetime.datetime.now()
         team_favorites_future = Favorite.query(Favorite.model_type == ModelType.TEAM, ancestor=user).fetch_async()
 
         live_events = EventHelper.getWeekEvents()
+        all_events_this_year_future = EventListQuery(now.year).fetch_async()
         favorite_team_keys = map(lambda f: ndb.Key(Team, f.model_key), team_favorites_future.get_result())
         favorite_teams_future = ndb.get_multi_async(favorite_team_keys)
 
+        future_events = filter(lambda e: e.start_date > now, all_events_this_year_future.get_result())
+        future_eventteams_futures = []
+        for event in future_events:
+            future_eventteams_futures.append(EventTeamsQuery(event.key_name).fetch_async())
```

Comment:
Is this fetching eventteams for all future events and then filtering for favorites? That sounds insanely inefficient. Perhaps try querying for eventteams of favorite teams, then querying for events that are in the future?